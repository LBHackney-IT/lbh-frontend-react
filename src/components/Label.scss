@import "node_modules/lbh-frontend/lbh/core/_typography.scss";
@import "node_modules/lbh-frontend/lbh/core/_label.scss";
