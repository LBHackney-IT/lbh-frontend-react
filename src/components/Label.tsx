import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import { Attributes, DataAttributes } from "../helpers/Attributes";

import "lbh-frontend/lbh/components/lbh-label/_label.scss";

/**
 * The property types for {@link Label}.
 *
 * This also supports all `aria-*` and `data-*` attributes.
 *
 * @noInheritDoc
 */
export interface LabelProps extends React.AriaAttributes, DataAttributes {
  id?: string;
  className?: string;
  labelFor?: string;
  children: React.ReactNode;
}

/**
 * A component for labelling inputs.
 */
export const Label: React.FunctionComponent<LabelProps> = (
  props: LabelProps
): React.ReactElement => {
  const { id, className, labelFor, children } = props;

  const extraAttributes = Attributes.ariaAndData(props);

  return (
    <label
      id={id}
      className={classNames("govuk-label", "lbh-label", className)}
      htmlFor={labelFor}
      {...extraAttributes}
    >
      {children}
    </label>
  );
};

Label.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  labelFor: PropTypes.string,
  children: PropTypes.node.isRequired
};
