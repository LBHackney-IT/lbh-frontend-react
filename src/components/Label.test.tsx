import React from "react";
import { create } from "react-test-renderer";

import { Label } from "./Label";

it("renders correctly with all props", () => {
  const component = create(
    <Label id="testId" className="class1 class2" labelFor="test">
      Test
    </Label>
  );

  expect(component).toMatchSnapshot();
});

it("renders correctly without optional props", () => {
  const component = create(<Label>Test</Label>);

  expect(component).toMatchSnapshot();
});
