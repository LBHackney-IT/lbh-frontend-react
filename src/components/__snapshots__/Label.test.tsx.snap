// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<label
  className="govuk-label lbh-label class1 class2"
  htmlFor="test"
  id="testId"
>
  Test
</label>
`;

exports[`renders correctly without optional props 1`] = `
<label
  className="govuk-label lbh-label"
>
  Test
</label>
`;
